lib/internal/process.js

```diff
@@ -101,6 +102,35 @@ function setup_hrtime() {
   };
 }
 
+function setupResourceUsage() {
+  const resourceUsage_ = process.resourceUsage;
+  const resValues = new Float64Array(18);
+
+  process.resourceUsage = function resourceUsage() {
+    resourceUsage_(resValues);
+    return {
+      userCpuTimeUsedSec: resValues[0],
+      userCpuTimeUsedMs: resValues[1],
+      systemCpuTimeUsedSec: resValues[2],
+      systemCpuTimeUsedMs: resValues[3],
+      maxResidentSetSize: resValues[4],
+      sharedMemSize: resValues[5],
```

Comment:
Would `integralSharedMemorySize` be more consistent with the next two values? http://man7.org/linux/man-pages/man2/getrusage.2.html